### PR TITLE
add go template logic to handle both fortio and nighthawk test setup

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/21161

### CODE REVIEW

**Comment by carolynhu** [comment]
cc @oschaaf Please take a look.

**Comment by carolynhu** [comment]
Here is the generated yaml files for fortio and nighthawk.
[testsetup_generated_yaml.zip](https://github.com/istio/tools/files/4549658/testsetup_generated_yaml.zip)

Tested locally, both nighthawk and fortio path work right now.

**Comment by carolynhu** [comment]
`benchmark_check_tools` is running in perf pipeline, so that we can ignore it's failures for now. Next PR aims to fix perf pipeline issue.

**Comment by oschaaf** [comment]
@carolynhu LGTM

**Comment by carolynhu** [comment]
@howardjohn if later on we want to remove fortio, just remove all the newly added thing in `fortio.yaml` in this PR. Since the base is for nighthawk.

File: perf/benchmark/values.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ appresources1:
 
 fortioImage: fortio/fortio:latest
 curlImage: tutum/curl:trusty
-nighthawkImage: envoyproxy/nighthawk-dev:latest
+nighthawkImage: envoyproxy/nighthawk-dev:59683b759eb8f8bd8cce282795c08f9e2b3313d4
```
Comment (richardwxn) [inline perf/benchmark/values.yaml:18]: shouldn''t we point to latest?

Comment (carolynhu) [reply perf/benchmark/values.yaml:18]: not yet, we have discussed this with Otto.

**Pull request merged**